src/opam_publish.ml

```diff
@@ -959,6 +972,12 @@ let repo_name =
          ~doc:"Local name of the repository to use (see the $(b,repo) \
                subcommand)")
 
+let repo_subdirectory =
+  Arg.(value & opt (some string) None & info ["subdirectory"]
```

Comment:
I'd rather give it more expressive name, to not be mistaken with package to be published subdirectory (archive, github, etc.), maybe `--opam-repo-subdir` or `--repo-packages-dir` (not completely relevant this one).

Reply:
OK, changing to `--opam-repo-subdir`.